nanofi/include/cxx/C2CallbackAgent.h

```diff
@@ -70,6 +69,7 @@ class C2CallbackAgent : public c2::C2Agent {
     c2_ag_stop_callback *stop;
 
  private:
+    utils::ThreadPool<utils::TaskRescheduleInfo> thread_pool_;
```

Comment:
Since this class inherits from C2Agent, it should also inherit its `ThreadPool`, so there should be no need for 2 `ThreadPool`s in `C2CallbackAgent`.

Do you think it would make sense to revert C2Agent to take the thread pool as a constructor argument instead of creating it, in the spirit of the "[Tell, Don't Ask](https://www.martinfowler.com/bliki/TellDontAsk.html)" principle, the "[Single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)" (object creation is a reponsibility) and "[Inversion of control](https://en.wikipedia.org/wiki/Inversion_of_control)"? This approach would solve the issue by making it possible to reuse the same thread pool for both classes, while leaving option to use separate thread pools if needed.

Reply:
Oh well, the thread_pool_ shouldn't be here at all. This file requires no changes. I forgot to remove it from the previous commit.